src/recval/metrics/metric_interface.py

```diff
@@ -1,42 +1,60 @@
-from abc import ABCMeta, abstractmethod
 from dataclasses import dataclass, field
 
-import pandas as pd
+import pandas
 from strenum import LowercaseStrEnum
 
 from recval.constants import DEFAULT_ITEM_COL, DEFAULT_USER_COL
 from recval.utils import MetaEnum
 
 
 @dataclass
-class MetricInterface(metaclass=ABCMeta):
+class MetricInterface:
     """Interface for an evaluation metric.
 
     Attributes:
-        df_hit (pd.DataFrame): dataframe of recommendation hits, sorted by col_user and rank,
-        df_hit_count (pd.DataFrame): dataframe of hit counts vs actual relevant items per user
-        cutoff (int): cutoff used to compute the recommendation
         col_user (str, optional): column name for user. Defaults to user_id.
         col_item (str, optional): column name for item. Defaults to item_id.
     """
 
-    df_hit: pd.DataFrame
-    df_hit_count: pd.DataFrame
-    cutoff: int
     col_user: str = DEFAULT_USER_COL
     col_item: str = DEFAULT_ITEM_COL
     name: str = field(init=False)
 
     def __post_init__(self) -> None:
-        self.name = self._name()
+        self.name = self.name_()
 
-    def _name(self) -> str:
+    @staticmethod
+    def name_() -> str:
```

Comment:
what does it mean having the underscore at the end of a method name?

Reply:
Se lo metti all'inizio rendi il metodo privato, poi pylint si lamenta se cerci di accederci da fuori la classe.